### PR TITLE
Fixes #4427 - Retried request duplicates cookies.

### LINKED ISSUE #4427: Retried HttpClient Requests can result in duplicates cookies
**Jetty version**
9.4.x

**Description**
`HttpClient` retries request only in one case: when the idle timeout is just expired on a connection while the request is about to be sent.
The implementation detects that the connection will be closed, and retries the request on another connection.

However, the retry runs `HttpConnection.normalizeRequest()` again, causing the cookies (and possibly other headers) to be duplicated.

### PR DESCRIPTION
**Pull request by sbordet**
#4427.

Introduced HttpRequest.normalized() to test and set whether
the request has already been normalized.
Added test case and few cleanups.

Signed-off-by: Simone Bordet <simone.bordet@gmail.com>

### CODE REVIEW

**Comment by joakime** [comment]
What happens if the server response uses a `Location` that isn't normalized? or isn't absolute?
Would the normalization now not occur in these cases?

A relative location ...

```
HTTP/1.1 301 Moved Permanently
Location: ../../alternate/path/
```

A schemeless location ...

```
HTTP/1.1 301 Moved Permanently
Location: //machine.com/alternate/path/
```

A non-normalized location ...

```
HTTP/1.1 301 Moved Permanently
Location: https://machine.com/static/../alternate/path/
```

**Comment by sbordet** [comment]
@joakime I don't understand? 
This is not URI normalization.
And a redirect will generate a new request object, so it's not affected by the issue.

**Comment by joakime** [comment]
Ah, never mind, I had a brain fart. 
I saw "Retried" as "Redirect" and the normalization code looked familiar.

**Review by joakime — APPROVED** [review]
LGTM, and now I understand the retry logic a bit better.

File: jetty-client/src/main/java/org/eclipse/jetty/client/HttpDestination.java [hunk 1]
```diff
@@ -326,10 +326,10 @@ private void process()
         }
     }
 
-    public boolean process(final Connection connection)
+    public boolean process(Connection connection)
     {
         HttpClient client = getHttpClient();
-        final HttpExchange exchange = getHttpExchanges().poll();
```
Comment (gregw) [inline jetty-client/src/main/java/org/eclipse/jetty/client/HttpDestination.java:332]: These finals were required to get rid of  code style warnings.   Why are you taking them out again?
Even if we don't want them, don't make unrelated changes in a PR.

**Review by gregw — APPROVED** [review]
I'd rather you didn't take out the finals that were put in to reduce the warnings. But the substance of the change is good.

File: jetty-client/src/main/java/org/eclipse/jetty/client/HttpDestination.java (hunk 1 above)
Comment (sbordet) [reply jetty-client/src/main/java/org/eclipse/jetty/client/HttpDestination.java:332]: What warnings? 
They are unnecessary and clutter the code, so while I touched the file I got rid of them.
